Cargo.toml

```diff
@@ -7,22 +7,23 @@ authors = [
   "Daniel P. Clark <6ftdan@gmail.com>"
 ]
 edition = "2021"
-
 description = "The tie between Ruby and Rust."
 repository = "https://github.com/danielpclark/rutie"
 readme = "README.md"
 keywords = ["cruby", "ruby", "rutie"]
 license = "MIT"
-
 build = "build.rs"
-links = "ruby"
 
 [features]
 no-link = [] # tells build.rs to not link to libruby
 
 [dependencies]
 libc = "0.2.121"
 lazy_static = "1.4.0"
+rb-sys = { version = "0.9.64", features = ["link-ruby"] }
```

Comment:
So we should not link-ruby by default (since gems rely on dynamic symbol lookup). Really only needs to be used if you are embedding libruby (i.e. for `cargo test` or a tool where you run `ruby_setup()` manually).

Here's how we have it configured in magnus, which offers an `embed` feature for linking ruby:

```toml
[features]
embed = ["rb-sys/link-ruby"]
rb-sys-interop = []
ruby-static = ["rb-sys/ruby-static"]

[dependencies]
rb-sys = { version = "0.9.65", default-features = false, features = [] }

[dev-dependencies]
rutie = { path = ".", features = ["embed", "rb-sys-interop"] }
```
```

Reply:
Thanks @ianks !

I implemented it initially this way because Rutie was/is doing the opposite linking by default and disabling linking via cargo feature flags. I implemented `magnus`'s approach as you suggested. 
I did not use the `rb-sys-interop` cargo feature flag tho, seemed like a magnus specific thing. 

Correct me if I am wrong but I don't think this is a breaking change for rutie users since is a relaxation on the constraint of having a Ruby installation built with `--enable-shared`?